commons/all/src/main/java/org/infinispan/commons/api/query/Query.java

```diff
@@ -0,0 +1,132 @@
+package org.infinispan.commons.api.query;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.commons.util.CloseableIterator;
+
+public interface Query<T> extends Iterable<T> {
+
+   /**
+    * Returns the Ickle query string.
+    */
+   String getQueryString();
+
+   /**
+    * Returns the results of a search as a list.
+    *
+    * @return list of objects that were found from the search.
+    */
+   List<T> list();
+
+   /**
+    *  Executes the query (a SELECT statement). Subsequent invocations cause the query to be re-executed.
+    *  <p>
+    *  Executing a DELETE is also allowed. No results will be returned in this case but the number of affected entries
+    *  will be returned as the hit count in the {@link QueryResult}.
+    *
+    * @return {@link QueryResult} with the results.
+    * @since 11.0
```

Comment:
Remove the @since from this method.
Add a @since 15.0 at the top of the interface